dialogue-target/src/main/java/com/palantir/dialogue/ExceptionDeserializerArgs.java

```
@@ -0,0 +1,132 @@
+        private boolean buildInvoked = false;
+        private @Nullable TypeMarker<T> returnType;
+        private final ImmutableMap.Builder<String, ErrorExceptionPair<?>> errorNameToExceptionTypeMarkers;
+
+        private Builder() {
+            this.errorNameToExceptionTypeMarkers = ImmutableMap.builder();
+        }
+
+        public Builder<T> returnType(TypeMarker<T> returnT) {
+            checkNotBuilt();
+            this.returnType = Preconditions.checkNotNull(returnT, "base type must be non-null");
+            return this;
+        }
+
+        /**
+         * Registers an error name with its corresponding error type and exception type.
+         * @param errorName the name of the error
+         * @param errorType the type marker for the error
+         * @param exceptionType the type marker for the exception. It is expected that the exception type implements a
+         * constructor that takes the error type as the first parameter, and an integer status code as the second.
+         */
+        public <U extends AbstractSerializableError<?>, V extends RemoteException> Builder<T> exception(
+                String errorName, TypeMarker<U> errorType, TypeMarker<V> exceptionType) {
+            checkNotBuilt();
+            this.errorNameToExceptionTypeMarkers.put(
+                    Preconditions.checkNotNull(errorName, "error name must be non-null"),
+                    new ErrorExceptionPair<>(
+                            Preconditions.checkNotNull(errorType, "error type must be non-null"),
+                            Preconditions.checkNotNull(exceptionType, "exception type must be non-null")));
+            return this;
+        }
+
+        public ExceptionDeserializerArgs<T> build() {
+            checkNotBuilt();
+            Preconditions.checkNotNull(returnType, "base type must be set");
+            buildInvoked = true;
+            return new ExceptionDeserializerArgs<>(returnType, errorNameToExceptionTypeMarkers.buildKeepingLast());
```

What do you think about
```suggestion
            return new ExceptionDeserializerArgs<>(returnType, errorNameToExceptionTypeMarkers.buildOrThrow());
```
? (to avoid silently dropping values, since it's never correct to register multiple bindings for a give error)

---

updated.